python/lsst/ip/isr/linearize.py

```diff
@@ -51,43 +48,65 @@ class Linearizer(abc.ABC):
             - one column for each image value
         To avoid copying the table the last index should vary fastest
         (numpy default "C" order)
-    detector : `lsst.afw.cameraGeom.Detector`
-        Detector object
-    override : `bool`, optional
-        Override the parameters defined in the detector/amplifier.
+    detector : `lsst.afw.cameraGeom.Detector`, optional
+        Detector object.  Passed to self.fromDetector() on init.
     log : `lsst.log.Log`, optional
         Logger to handle messages.
+    kwargs : `dict`, optional
+        Other keyword arguments to pass to the parent init.
 
     Raises
     ------
     RuntimeError :
         Raised if the supplied table is not 2D, or if the table has fewer
         columns than rows (indicating that the indices are swapped).
-    """
 
-    _OBSTYPE = "linearizer"
-    """The dataset type name used for this class"""
+    Notes
+    -----
+    The linearizer attributes stored are:
+
+    hasLinearity : `bool`
+        Whether a linearity correction is defined for this detector.
+    override : `bool`
+        Whether the detector parameters should be overridden.
+    ampNames : `list` [`str`]
+        List of amplifier names to correct.
+    linearityCoeffs : `dict` [`str`, `numpy.array`]
+        Coefficients to use in correction.  Indexed by amplifier
+        names.  The format of the array depends on the type of
+        correction to apply.
+    linearityType : `dict` [`str`, `str`]
```

Comment:
Having this as a dictionary would imply that, in principle, each amp would have a different linearity type. But i don't think that's currently the case, correct?

Reply:
Each amp can have a different linearity type, and the correction will apply whatever is defined.  We currently don't have a way to measure different linearity types simultaneously in `cp_pipe`, but such a calibration could be constructed if necessary.  This is largely a carryover from the `cameraGeom` defined linearity parameters, which were amplifier based.